freeipa/src/main/java/com/sequenceiq/freeipa/service/telemetry/TelemetryConfigService.java

```diff
@@ -129,13 +130,28 @@ public Map<String, SaltPillarProperties> createTelemetryPillarConfig(Stack stack
     }
 
     private Map<String, SaltPillarProperties> getMonitoringPillarConfig(Stack stack, Telemetry telemetry, char[] passwordInput, boolean cdpSaasEnabled,
-            boolean computeMonitoring) {
+            boolean computeMonitoring) throws CloudbreakOrchestratorFailedException {
         Map<String, Object> config = new HashMap<>();
         Monitoring monitoring = telemetry.getMonitoring();
         if (monitoring != null && telemetry.isMonitoringFeatureEnabled()) {
             LOGGER.debug("Monitoring is enabled, filling configs ...");
+            String accessKey = null;
+            char[] privateKey = null;
+            if (monitoringConfigService.isMonitoringEnabled(cdpSaasEnabled, computeMonitoring)) {
+                try {
+                    MonitoringCredential monitoringCredential = altusMachineUserService.getOrCreateMonitoringCredentialIfNeeded(
+                            stackService.getStackById(stack.getId()));
+                    if (monitoringCredential != null) {
+                        accessKey = monitoringCredential.getAccessKey();
+                        privateKey = Optional.of(monitoringCredential.getPrivateKey()).map(String::toCharArray).orElse(null);
+                    }
+                } catch (IOException e) {
+                    throw new CloudbreakOrchestratorFailedException(e);
+                }
+            }
```

Comment:
why is this not part of the `monitoringConfigService.createMonitoringConfig` call?

Reply:
there are core/freeipa specific code there, most of these changes are duplications (similar that was done for  DatabusCredential) - but see my comment above as there is a refactor that happening at the moment (mainly focusing on the config generation)